### PR TITLE
Create Autoencoders.md

### LINKED ISSUE #1113: Autoencoders
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md).

### Description

Autoencoders are neural networks trained to reconstruct input data. They are used for dimensionality reduction, feature learning, and denoising. Autoencoders can learn latent representations of data that capture the underlying structure.

### Self-service

- [X] I'd be willing to address this documentation request myself.

### PR DESCRIPTION
**Pull request by inkerton**
### Description
Autoencoders are a type of artificial neural network used to learn efficient representations of data in an unsupervised manner. Their goal is to map input data to a compressed representation (encoding) and then reconstruct the input data from this representation (decoding). Autoencoders are widely used for dimensionality reduction, feature learning, and data denoising.

Fixes #1113 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/machine-learning/Autoencoders.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+- **Latent Space Representation**:  
+  The compressed representation in the latent space can be used for other tasks like clustering, visualization, or as input for another machine learning model.
+
+### Split Criteria:
+Autoencoders are trained to minimize the reconstruction loss between the input and the output, so the criteria are not based on splitting but on reconstructing the input accurately.
+
+### Time Complexity:
+- **Training Complexity**:  
+  The time complexity depends on the number of layers and neurons in the encoder and decoder. For an autoencoder with \( l \) layers and \( n \) neurons per layer, the complexity is \( O(n \cdot l) \).
```
Comment (ajay-dhangar) [inline docs/machine-learning/Autoencoders.md]: replace `\( O(n \cdot l) \)` with `$O(n \cdot l)$`, `\( l \)` with `$l$`, and `\( n \)` with `$n$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by inkerton** [comment]
@ajay-dhangar Done all

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**